utilities/transactions/transaction_base.cc

```diff
@@ -215,8 +217,16 @@ Status TransactionBaseImpl::Get(const ReadOptions& read_options,
 Status TransactionBaseImpl::GetForUpdate(const ReadOptions& read_options,
                                          ColumnFamilyHandle* column_family,
                                          const Slice& key, std::string* value,
-                                         bool exclusive) {
-  Status s = TryLock(column_family, key, true /* read_only */, exclusive);
+                                         bool exclusive,
+                                         const bool do_validate) {
+  assert(do_validate || read_options.snapshot == nullptr);
+  if (UNLIKELY(!do_validate && read_options.snapshot != nullptr)) {
```

Comment:
As always, I'm not a fun of UNLIKELY hint. Do we really need it?

Reply:
I have seen cases that it improves performance and here it might too. I do not see it to be hurtful in anyway.

Reply:
It is hurtful because it tries to dictate compiler behavior unnecessarily. It's just like that we don't encourage people to deep tune RocksDB behavior unless it's needed, I am not a fan of doing it to compiler either.

Furthermore, just have these 8 macros inside the code for no reason makes it less readable. Imagine someone grabs RocksDB source code and read this function, and sees a "UNLIKELY()".

Reply:
I believe it is a "hint" rather than "dictation". And it is not "unnecessary" as the previous benchmarks has shown. I actually find them to make the code more readable if they are used properly. Here is a proper use case as the if-then branch is not taken unless there is a user bug.

Reply:
I even feel that we should take the hit even we see some minor regression.  "Previous benchmark" is even weaker argument. It's more readable to you because you wrote so many UNLIKELY() in the code base so that you are so familiar with what it is for when you see it. It's not the case for someone new to the code base. In fact, I'm always reluctant to the use of macros in general. This is also the rule written in the Google C++ Style we are following: https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros

Reply:
I am going to remove UNLIKELY here as I do not want this minor point to become a blocker for this PR.

Reply:
@maysamyabandeh I appreciate it.